test/integration/esys-policy-password.int.c

```diff
@@ -0,0 +1,304 @@
+                 .buffer = {},
+             },
+        },
+    };
+
+    TPM2B_AUTH authValuePrimary = {
+        .size = 5,
+        .buffer = {1, 2, 3, 4, 5}
+    };
+
+    TPM2B_SENSITIVE_CREATE inSensitivePrimary = {
+        .size = 4,
+        .sensitive = {
+            .userAuth = {
+                 .size = 0,
+                 .buffer = {0 },
+             },
+            .data = {
+                 .size = 0,
+                 .buffer = {0},
+             },
+        },
+    };
+    return 0;
+
+    inSensitivePrimary.sensitive.userAuth = authValuePrimary;
+
+    TPM2B_DATA outsideInfo = {
+        .size = 0,
+        .buffer = {},
+    };
+
+    TPML_PCR_SELECTION creationPCR = {
+        .count = 0,
+    };
+
+    TPM2B_AUTH authValue = {
+        .size = 0,
+        .buffer = {}
+    };
+
+    r = Esys_TR_SetAuth(esys_context, ESYS_TR_RH_OWNER, &authValue);
+    goto_if_error(r, "Error: TR_SetAuth", error);
+
+    ESYS_TR primaryHandle_handle;
+    TPM2B_PUBLIC *outPublic;
+    TPM2B_CREATION_DATA *creationData;
+    TPM2B_DIGEST *creationHash;
+    TPMT_TK_CREATION *creationTicket;
+
+    r = Esys_CreatePrimary(esys_context, ESYS_TR_RH_OWNER, ESYS_TR_PASSWORD,
+                           ESYS_TR_NONE, ESYS_TR_NONE, &inSensitivePrimary, &inPublic,
+                           &outsideInfo, &creationPCR, &primaryHandle_handle,
+                           &outPublic, &creationData, &creationHash,
+                           &creationTicket);
+    goto_if_error(r, "Error esys create primary", error);
+
+    ESYS_TR policySession;
+    TPMT_SYM_DEF policySymmetric = {.algorithm = TPM2_ALG_AES,
+                                    .keyBits = {.aes = 128},
+                                    .mode = {.aes = TPM2_ALG_CFB}
+    };
+    TPM2B_NONCE *policyNonceTpm;
+    TPM2B_NONCE policyNonceCaller = {
+        .size = 20,
+        .buffer = {11, 12, 13, 14, 15, 16, 17, 18, 19, 11,
+                   21, 22, 23, 24, 25, 26, 27, 28, 29, 30}
+    };
+    r = Esys_StartAuthSession(esys_context, ESYS_TR_NONE, ESYS_TR_NONE,
+                              ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+                              &policyNonceCaller,
+                              TPM2_SE_POLICY, &policySymmetric, TPM2_ALG_SHA1, &policySession,
+                              &policyNonceTpm);
+    goto_if_error(r, "Error: During initialization of policy trial session", error);
+
+    r = Esys_PolicyPassword(
+        esys_context,
+        policySession,
+        ESYS_TR_NONE,
+        ESYS_TR_NONE,
+        ESYS_TR_NONE
+        );
+    goto_if_error(r, "Error: PolicyAuthValue", error);
+
+    r = Esys_TR_SetAuth(esys_context, primaryHandle_handle, &authValuePrimary);
+    goto_if_error(r, "Error: TR_SetAuth", error);
+
+    TPM2B_AUTH authKey2 = {
+        .size = 6,
+        .buffer = {6, 7, 8, 9, 10, 11}
+    };
+
+    TPM2B_SENSITIVE_CREATE inSensitive2 = {
+        .size = 1,
+        .sensitive = {
+            .userAuth = {
```

Comment:
that probably not needed since it will be set to authKey2 just below.

Reply:
done